File: RKGeonames/DemographicDataClient.m
//
//  DemographicHTTPClient.m
//  RKGeonames
//
//  Created by Stefan Buretea on 4/2/14.
//  Copyright (c) 2014 Stefan Burettea. All rights reserved.
//

#import "DemographicDataClient.h"

#import "CountryData.h"
#import "ManagedObjectStore.h"


@implementation DemographicDataClient

SingletonImplemetion(DemographicDataClient);

static NSString *const TOTAL_POPULATION_INDICATOR_STRING = @"SP.POP.TOTL";
static NSString *const POPULATION_GROWTH_INDICATOR_STRING = @"SP.POP.GROW";
static NSString *const BIRTH_RATE_INDICATOR_STRING = @"SP.DYN.CBRT.IN";
static NSString *const DEATH_RATE_INDICATOR_STRING = @"SP.DYN.CDRT.IN";
- (void) getDataForCountry:(NSString *)country
{    
    // when no data is stored locally, get it from the net
    if([self.delegate respondsToSelector:@selector(updateView:)]){
        [self.delegate updateView:self];
    }    
}


@end

File: RKGeonames/DemographicDataClient.swift
//
//  DemographicDataClient.swift
//  RKGeonames
//
//  Created by Stefan Buretea on 9/24/14.
//  Copyright (c) 2014 Stefan Burettea. All rights reserved.
//

import Foundation

class RKGDemographicDataClient: NSObject {
    
    class var TOTAL_POPULATION_INDICATOR_STRING: NSString {
        return "SP.POP.TOTL" as NSString
    }
    
    class var POPULATION_GROWTH_INDICATOR_STRING: NSString {
        return "SP.POP.GROW"
    }
    
    class var BIRTH_RATE_INDICATOR_STRING: NSString {
        return "SP.DYN.CBRT.IN"
    }
    
    class var DEATH_RATE_INDICATOR_STRING: NSString {
        return "SP.DYN.CDRT.IN"
    }

    class var sharedInstance: RKGDemographicDataClient {
    struct Static {
        static var onceToken: dispatch_once_t = 0
        static var instance: RKGDemographicDataClient? = nil
        }
        
        dispatch_once(&Static.onceToken) {
            Static.instance = RKGDemographicDataClient()
        }
        
        return Static.instance!
    }
    
    func getDataForCountry(country: NSString, completion: () -> Void) {
        completion()
    }
}